File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.c
bool canMakeSubsequence(char *str1, char *str2) {
    int i = 0, j = 0;
    int len1 = strlen(str1), len2 = strlen(str2);

    while (i < len1 && j < len2) {
        // Check for direct match or single cyclic increment
        if (str1[i] == str2[j] || (str1[i] - 'a' + 1) % 26 + 'a' == str2[j]) {
            j++;
        }
        i++;
    }

    // Return true only if all characters in str2 are matched
    return j == len2;
}

File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int j = 0;
        for (int i = 0; i < str1.length() && j < str2.length(); i++) {
            // Check for direct match or single cyclic increment
            if (str1.charAt(i) == str2.charAt(j) || 
                (char)((str1.charAt(i) - 'a' + 1) % 26 + 'a') == str2.charAt(j)) {
                j++;
            }
        }
        
        // Return true only if all characters in str2 are matched
        return j == str2.length();
    }
}